prow/config/config.go

```diff
@@ -1243,25 +1262,112 @@ type Branding struct {
 	HeaderColor string `json:"header_color,omitempty"`
 }
 
+// Note RerunAuthConfigs is a deprecated type
 // RerunAuthConfigs represents the configs for rerun authorization in Deck.
 // Use `org/repo`, `org` or `*` as key and a `RerunAuthConfig` struct as value.
 type RerunAuthConfigs map[string]prowapi.RerunAuthConfig
 
-// GetRerunAuthConfig returns the appropriate RerunAuthConfig based on the provided Refs.
-func (rac RerunAuthConfigs) GetRerunAuthConfig(refs *prowapi.Refs) prowapi.RerunAuthConfig {
-	if refs == nil || refs.Org == "" {
-		return rac["*"]
+type DefaultRerunAuthConfigEntry struct {
+	// Matching/filtering fields. All filters must match for an entry to match.
+
+	// OrgRepo matches against the "org" or "org/repo" that the presubmit or postsubmit
+	// is associated with. If the job is a periodic, extra_refs[0] is used. If the
+	// job is a periodic without extra_refs, the empty string will be used.
+	// If this field is omitted all jobs will match.
+	OrgRepo string `json:"repo,omitempty"`
+	// Cluster matches against the cluster alias of the build cluster that the
+	// ProwJob is configured to run on. Recall that ProwJobs default to running on
+	// the "default" build cluster if they omit the "cluster" field in config.
+	Cluster string `json:"cluster,omitempty"`
+
+	// Config is the RerunAuthConfig to apply if the filter fields all match the
+	// ProwJob. Note that when multiple entries match a ProwJob the entry with the
+	// highest specification is used.
+	Config *prowapi.RerunAuthConfig `json:"rerun_auth_configs,omitempty"`
+}
+
+func (d *Deck) GetRerunAuthConfig(refs *prowapi.Refs, cluster string) *prowapi.RerunAuthConfig {
+	var config *prowapi.RerunAuthConfig
+
+	var priority func(drac *DefaultRerunAuthConfigEntry) bool
+	if refs == nil {
```

Comment:
Sorry that I have overlooked previously. I believe we'll also need to consider cluster here, and taken a look at the `match` function we could do something even simpler, this entire if else condition can be changed to:
```
var orgRepo string
if refs != nil {
  orgRepo = refs.OrgRepoString()
}

for _, drac := range d.DefaultRerunAuthConfigs {
		if drac.matches(orgRepo, cluster) {
			config = drac.Config
		}
	}
```

Reply:
Modified!